kube/tests/find.rs

```diff
@@ -0,0 +1,91 @@
+use kube::{Error, Result};
+use serde_json::Value;
+use std::iter;
+
+const MINIMAL_OBJECT: &str = r#"{
+	"kind": "Deployment",
+	"apiVersion": "apps/v1"
+}"#;
+
+type Return = Result<Vec<Value>>;
+
+fn find_from(text: &str) -> Return {
+	let val: Value = serde_json::from_str(text).unwrap();
+	kube::find(&val)
+}
+
+fn assert_invalid(err: Return) {
+	assert!(err.is_err());
+	assert_eq!(err.unwrap_err(), Error::Invalid);
+}
+
+fn assert_valid(ok: Return, times: usize) {
+	assert!(ok.is_ok());
+
+	let object = serde_json::from_str(MINIMAL_OBJECT).unwrap();
+	let objects: Vec<Value> = iter::repeat(object).take(times).collect();
+	assert_eq!(ok.unwrap(), objects)
+}
+
+#[test]
+fn finds_objects() {
+	let json = MINIMAL_OBJECT;
+	let found = find_from(&json);
+	assert_valid(found, 1);
+
+	let json = format!(r#"{{"a":{0}, "b":{0}}}"#, MINIMAL_OBJECT);
+	let found = find_from(&json);
+	assert_valid(found, 2);
+
+	let json = format!(
+		r#"{{"a":{{ "b": {0}, "c": {{ "d": {0}}}, "e": {0}}}}}"#,
+		MINIMAL_OBJECT
+	);
+	let found = find_from(&json);
+	assert_valid(found, 3);
+}
+
+#[test]
+fn disallow_unclear_paths() {
+	let json = format!("[{0}, {0}]", MINIMAL_OBJECT);
+	let found = find_from(&json);
+	assert_invalid(found);
+
+	let json = format!(r#"{{"a":{0}, "b":[{0}, {0}]}}"#, MINIMAL_OBJECT);
+	let found = find_from(&json);
+	assert_invalid(found);
+
+	let json = format!(r#"{{"a":{{ "b": {0}, "c": [{0}, {0}]}}}}"#, MINIMAL_OBJECT);
+	let found = find_from(&json);
+	assert_invalid(found);
+}
+
+#[test]
+fn disallow_primitives() {
```

Comment:
Is there some way to make this test `parametrized` like this [feature](https://docs.pytest.org/en/stable/parametrize.html) from `pytest`?

Reply:
We would need to use another test framework than the built-in, a candidate is the [rstest](https://crates.io/crates/rstest) crate - 
 which is similar to pytest. My experience with Rust is the content from the beginning of The Book until Chapter 15, which is extremely limited; therefore, I prefer not to use alternative test frameworks until I have a better grasp of the language and its ecosystem.

Changing from blocks of code to a vector with all the test cases would probably be enough to remove the repetition that concerns you.